addons/web/static/src/views/list/list_renderer.js

```diff
@@ -604,6 +605,39 @@ export class ListRenderer extends Component {
             if (type !== "integer" && type !== "float" && type !== "monetary") {
                 continue;
             }
+            let currencyId;
+            if (type === "monetary") {
+                const currencyField =
+                    this.fields[column.name].currency_field ||
+                    this.props.list.activeFields[column.name].options.currency_field ||
+                    ("x_currency_id" in this.props.list.activeFields
+                        ? "x_currency_id"
+                        : "currency_id");
```

Comment:
Not sure about this part. In formatters.js, we honor the option first, which makes sense, because otherwise you can't override the field definition. Also, I can't find anywhere in the codebase where we handle x_currency_id, so I would not deal with this here.